### PR TITLE
Add WebUI/API endpoints, documentation and job to copy a staging project

### PR DESCRIPTION
**Pull request by dmarcoux**
copy
For the WebUI part, I have created a view to preview the copy of a staging project. Right now, it's quite basic and rough, but I prefer to gather feedback before going further (if we decide it's worth it). Anyway, my idea is to provide some information about what is to be copied instead of just copying a staging project without giving any feedback to the user (this would be so if we used a modal for example).

Let me know what you think. Here's how it looks:

Clickable icons to reach the `Copy` page, they are displayed when we are configuring the staging workflow (so after clicking the `Configure` on the staging workflow show page):
icon

The `Copy` page:
copy

### CODE REVIEW

**Comment by dmarcoux** [comment]
["create staging project" from a template](https://trello.com/c/diFikqV6/1018-create-staging-project-from-a-template)

File: src/api/app/jobs/staging_project_copy_job.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class StagingProjectCopyJob < ApplicationJob
+  def perform(staging_workflow_project_name, original_staging_project_name, staging_project_copy_name)
```
Comment (bgeuken) [inline src/api/app/jobs/staging_project_copy_job.rb]: Nitpick: I would think about shortening the variable names.

**Comment by bgeuken** [comment]
> Let me know what you think.

I agree, but well... I was part of this discussion:wink:

File: src/api/app/jobs/staging_project_copy_job.rb (hunk 1 above)
Comment (dmarcoux) [reply src/api/app/jobs/staging_project_copy_job.rb]: I agree that they are quite long, but I didn't find something shorter without losing clarity.

**Comment by dmarcoux** [comment]
Specs are flickering hard...

**Comment by Ana06** [comment]
@dmarcoux 

> Specs are flickering hard...

It seems related with Rantly and not with your code. I am on it.

**Comment by dmarcoux** [comment]
Thank you @Ana06 for having a look.

File: src/api/app/jobs/staging_project_copy_job.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    staging_workflow_project = Project.find_by!(name: staging_workflow_project_name)
+    original_staging_project = staging_workflow_project.staging.staging_projects.find_by!(name: original_staging_project_name)
+    original_staging_project.copy(staging_project_copy_name)
```
Comment (dmarcoux) [inline src/api/app/jobs/staging_project_copy_job.rb]: FYI: If this fails, the user has no way to know what happened.

Is there something we can do about this?

Comment (bgeuken) [reply src/api/app/jobs/staging_project_copy_job.rb]: I'll have a look into that

File: src/api/app/jobs/staging_project_copy_job.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/jobs/staging_project_copy_job.rb]: How about this?:

staging_workflow_project_name -> workflow_name
original_staging_project_name -> original_name
staging_project_copy_name -> copy_name

File: src/api/app/views/webui2/webui/staging/projects/preview_copy.html.haml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+.row
+  .col-xl-10
+    .card.mb-3
+      = render(partial: 'webui2/webui/project/tabs', locals: { project: @staging_project })
+      .card-body
+        %h3= @pagetitle
+        %p The following data will be copied:
+        %ul
+          %li Attributes
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/staging/projects/preview_copy.html.haml]: Attributes are not copied

File: src/api/app/views/webui2/webui/staging/projects/preview_copy.html.haml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          %li Groups
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/staging/projects/preview_copy.html.haml]: Let's combine this with user config: `Users & Groups`

File: src/api/app/jobs/staging_project_copy_job.rb (hunk 2 above)
Comment (dmarcoux) [reply src/api/app/jobs/staging_project_copy_job.rb]: I just pushed a commit for a possible solution. It's not working yet as explained in the commit, but we can discuss about this.

File: src/api/app/jobs/staging_project_copy_job.rb (hunk 1 above)
Comment (dmarcoux) [reply src/api/app/jobs/staging_project_copy_job.rb]: Hum... I prefer longer names, they are clearer. If someone else thinks we should go for the names you suggest, I'm fine with changing them.

File: src/api/app/jobs/staging_project_copy_job.rb (hunk 2 above)
Comment (dmarcoux) [reply src/api/app/jobs/staging_project_copy_job.rb]: I removed the commit from the PR. I will create another PR when this one is merged.

**Pull request merged**